snap/snapcraft.yaml

```diff
@@ -1,222 +1,230 @@
+    command: opt/kafka/bin-wrapper.bash
     plugs:
       - network
       - network-bind
     environment:
-      _wrapper_script: kafka-verifiable-consumer.sh
+      bin_script: kafka-verifiable-consumer.sh
   verifiable-producer:
-    command: script-wrapper.bash
+    command: opt/kafka/bin-wrapper.bash
     plugs:
       - network
       - network-bind
     environment:
-      _wrapper_script: kafka-verifiable-producer.sh
+      bin_script: kafka-verifiable-producer.sh
   trogdor:
-    command: script-wrapper.bash
+    command: opt/kafka/bin-wrapper.bash
     plugs:
       - network
       - network-bind
     environment:
-      _wrapper_script: trogdor.sh
-  exporter:
-    command: exporter-wrapper.bash
-    daemon: simple
-    install-mode: disable
-    plugs:
-      - network
-      - network-bind
+      bin_script: trogdor.sh
 
 parts:
-  dependencies:
+  kafka:
     plugin: nil
+    source: https://downloads.apache.org/kafka/${SNAPCRAFT_PROJECT_VERSION}/kafka_2.13-${SNAPCRAFT_PROJECT_VERSION}.tgz
     stage-packages:
     - openjdk-17-jre-headless
+    - util-linux
     - libpsl5
     - curl
+    override-build: |
+      snapcraftctl build
+      cp -r ./* $SNAPCRAFT_PART_INSTALL
     override-prime: |
       snapcraftctl prime
       rm -vf usr/lib/jvm/java-17-openjdk-amd64/lib/security/cacerts
-  kafka:
-    plugin: nil
-    override-build: |
-      archive="kafka_2.13-${SNAPCRAFT_PROJECT_VERSION}.tgz"
-      curl -o "${archive}" "https://downloads.apache.org/kafka/${SNAPCRAFT_PROJECT_VERSION}/${archive}"
-      mkdir -p $CRAFT_PART_INSTALL/opt/kafka
-      tar -xzvf "${archive}" -C "${CRAFT_PART_INSTALL}/opt/kafka" --strip-components=1
-  kafka-exporter:
-    plugin: nil
-    after: [kafka]
-    override-build: |
-      exporter="kafka_exporter-1.6.0.linux-amd64.tar.gz"
-      curl -sLo "${exporter}" "https://github.com/danielqsj/kafka_exporter/releases/download/v1.6.0/${exporter}"
-      tar -xvf "${exporter}" --strip-components=1
-      cp kafka_exporter $CRAFT_PART_INSTALL/
-    organize:
-      kafka_exporter: opt/kafka/kafka_exporter
+      mkdir -p $SNAPCRAFT_PART_INSTALL/opt/kafka
   prometheus-exporter:
     plugin: nil
     after: [kafka]
     override-build: |
       curl -o jmx_prometheus_javaagent.jar "https://repo1.maven.org/maven2/io/prometheus/jmx/jmx_prometheus_javaagent/0.17.0/jmx_prometheus_javaagent-0.17.0.jar"
       EXPECTED_SHA="48ab23a7514f9de5d2ca0acbb8ed1573b3c2ecbef7c5dc4d37c4ba125e538199"
-      JAR_SHA=$(sha256sum jmx_prometheus_javaagent.jar | cut -d" " -f1)
+      JAR_SHA=$(sha256sum jmx_prometheus_javaagent.jar | cut -d " " -f1)
       if [ $JAR_SHA != $EXPECTED_SHA ]; then exit 1; fi
-      cp jmx_prometheus_javaagent.jar $CRAFT_PART_INSTALL/
+      cp jmx_prometheus_javaagent.jar $SNAPCRAFT_PART_INSTALL/
     organize:
-      jmx_prometheus_javaagent.jar: opt/kafka/extra/jmx_prometheus_javaagent.jar
+      jmx_prometheus_javaagent.jar: opt/kafka/jmx_prometheus_javaagent.jar
```

Comment:
**question** should we put the JMX jar in `$SNAP/usr/lib`? I mean, that seems more appropriate if also other components in the SNAP would eventually need this jar?

Reply:
The exporter is an 'opt-ional` package. It's accessible from any other package by referencing it's path

Reply:
JSYK, it's the same path as ZK - https://github.com/canonical/charmed-zookeeper-snap/blob/d8488a6b083ffed7a35fe6a93adb68e06160238b/snap/snapcraft.yaml#L67-L68